### DISCUSSION TITLE
[Question] : Does Flet Support bidirectional communation with Java api on android

### DISCUSSION THREAD

**Discussion by baseplate-admin** [flet-dev/flet#1652] [Q&A]
### Question

Hi, I was wondering since flet will eventually support android. If it was possible to invoke Java codes from Python and vice versa?

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by FeodorFitsner** [comment]
It won't be direct support for that, but Flet app is a normal Flutter app, so you'd be able to implement a Flutter plugin working with Java code and then add support to Flet by implementing Dart and Python parts.

**Reply by baseplate-admin** [reply]
> It won't be direct support for that, but Flet app is a normal Flutter app, so you'd be able to implement a Flutter plugin working with Java code and then add support to Flet by implementing Dart and Python parts.

Yep sounds what i did, basically i have [this](https://github.com/coreproject-moe/android-flutter/blob/c1d05ba1a1dea6391ce21e0ffef973a8d1c41ab9/android/app/src/main/kotlin/moe/coreproject/gecko/PlatformGeckoView.kt) and i was wondering how to implement it in `flet `

**Comment by baseplate-admin** [comment]
Hi @FeodorFitsner, whats the recommended way now? Does flet have Dart bridge (something like [platform-views](https://docs.flutter.dev/platform-integration/android/platform-views) )?
